databases/functions/topics/topic-get-topic-detail-by-id.js

```diff
@@ -0,0 +1,35 @@
+const {QueryTypes} = require('sequelize');
+
+const getTopicDetailById = async (sequelize, selfUserId, topicName) => {
+  const query = `SELECT 
+      topics.*, 
+      COUNT(user_topics.topic_id) as topic_follower_count, 
+      CASE COALESCE(user_following_topics.user_id, '') 
+        WHEN '' THEN false
+        ELSE true
+      END as is_following
+    FROM topics
+    LEFT JOIN user_topics
+      ON user_topics.topic_id = topics.topic_id
+    LEFT JOIN user_topics AS user_following_topics
+      ON user_following_topics.topic_id = topics.topic_id AND user_following_topics.user_id=:selfUserId
+    WHERE topics.name=:topicName
+    GROUP BY user_topics.topic_id, topics.topic_id, user_following_topics.user_id`;
+
+  try {
+    const response = await sequelize.query(query, {
+      type: QueryTypes.SELECT,
+      replacements: {
+        selfUserId,
+        topicName
+      }
+    });
+
+    return response[0];
+  } catch (error) {
+    console.log(error);
+    throw new Error(error);
+  }
+};
+
+module.exports = getTopicDetailById;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `getTopicDetailById` is well written and follows good practices. However, there's a potential issue with error handling. Currently, the error is logged to the console and then thrown again. This could lead to revealing sensitive information about your database structure or other internal details if these error messages are exposed to the end user. It would be better to throw a custom error message instead.

```diff
-    console.log(error);
-    throw new Error(error);
+    console.error('Error while fetching topic details:', error.message);
+    throw new Error('Failed to fetch topic details');
```
Also, it's important to note that this function will return `undefined` if the query doesn't match any records in the database because `response[0]` will be `undefined`. You might want to handle this case explicitly depending on your application's requirements.


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
will be fixed on next commit